### PR TITLE
sensors/vehicle_imu: Integrator split coning version and make header only

### PR DESCRIPTION
**Pull request by dagar**
- split integrator into specialized version with coning compensation
 - minor refactoring in preparation of using full FIFO data (where available)

### CODE REVIEW

File: src/modules/sensors/vehicle_imu/Integrator.hpp [hunk 1]
```diff
@@ -95,29 +83,106 @@ class Integrator
...
+	void put(const matrix::Vector3f &val, const float dt)
+	{
+		// Use trapezoidal integration to calculate the delta integral
+		const matrix::Vector3f delta_alpha{integrate(val, dt)};
+
+		// Calculate coning corrections
+		// Coning compensation derived by Paul Riseborough and Jonathan Challinger,
+		// following:
+		// Tian et al (2010) Three-loop Integration of GPS and Strapdown INS with Coning and Sculling Compensation
+		// Sourced: http://www.sage.unsw.edu.au/snap/publications/tian_etal2010b.pdf
```
Comment (bresch) [inline src/modules/sensors/vehicle_imu/Integrator.hpp:155]: This reference is really difficult (impossible?) to find. Should we change it with https://arc.aiaa.org/doi/pdf/10.2514/2.4228 that contains the same equations?

**Review by bresch — CHANGES_REQUESTED** [review]

File: src/modules/sensors/vehicle_imu/Integrator.hpp (hunk 1 above)
Comment (dagar) [reply src/modules/sensors/vehicle_imu/Integrator.hpp:155]: Might as well. FYI @priseborough

Comment (dagar) [reply src/modules/sensors/vehicle_imu/Integrator.hpp]: Done.

File: src/modules/sensors/vehicle_imu/Integrator.hpp [hunk 2]
```diff
@@ -95,29 +89,123 @@ class Integrator
...
-	uint32_t _reset_interval_min{1}; /**< the interval after which the content will be published and the integrator reset */
+protected:
 
-	uint8_t _integrated_samples{0};
+	matrix::Vector3f integrate(const matrix::Vector3f &val, const float dt)
+	{
+		// Use trapezoidal integration to calculate the delta integral
+		_integrated_samples++;
+		_integral_dt += dt;
+		const matrix::Vector3f delta_alpha{(val + _last_val) *dt * 0.5f};
```
Comment (bresch) [inline src/modules/sensors/vehicle_imu/Integrator.hpp:127]:
```suggestion
		const matrix::Vector3f delta_alpha{(val + _last_val) * dt * 0.5f};
```

**Review by bresch — APPROVED** [review]

**Comment by dagar** [comment]
Most of this was rolled into https://github.com/PX4/PX4-Autopilot/pull/17640 for further testing and improvements.

**Pull request closed without merging**